corehq/pillows/mappings/app_mapping.py

```diff
@@ -56,7 +55,11 @@
                                     'type': 'date'},
                 'description': {'type': 'string'},
                 'doc_type': {'index': 'not_analyzed', 'type': 'string'},
-                'domain': {'type': 'string'},
+                'domain': {'fields': {'domain': {'index': 'analyzed',
+                                                 'type': 'string'},
+                                      'exact': {'index': 'not_analyzed',
+                                                'type': 'string'}},
+                           'type': 'multi_field'},
```

Comment:
Both this change and the other will retain the same behavior for existing filters acting on `"domain"`, only `"domain.exact"` will use the non-analysed index.